### ISSUE TITLE
New version Sipjs - EarlyMedia

### ISSUE THREAD

**Issue by 7i77an** [InnovateAsterisk/Browser-Phone#141]
Do you have in the roadmap move to the latest version of sipjs with the new api or did you test earlyMedia with sipjs version 0.20?

I couldn't get it to work and I saw a comment you made where you were planning to hack this or update the sipjs version.
I saw your changes and you chose to do the hack.

Create an issue in sipjs referring to this topic (https://github.com/onsip/SIP.js/issues/937)

Sorry for placing this here if it is not adequate.

**Comment by InnovateAsterisk** [comment]
I took quite a lot time to consider updating to the latest version of SIPJS, but in the end couldn't justify the number of breaking changes it would create... for now. It seems that most of the improvements in the newer versions are already taken care of with workarounds or don't really effect the browser phone. The biggest push for development is around XMPP and Group Call/Chat.

I did however take a look at EarlyMedia, and made some changes in the 183 session handling, that was previously not enabled or used. If you send back a 180 status, the phone will continue as it did before and play its own early media, however if you send back a 183, it will now attempt to open the media path, and play back any audio stream using a provisional answer. 

In my tests I found that either Asterisk or SIPJS failed to use rel100 (the mechanism to reliably transmit messages in the 1XX response code ranges), and so I have disabled rel100 (or left it as disabled). rel100 doesn't appear to be a prerequisite for early media, only a nice to have... so for now its off.
